aiohttp/client.py

```diff
@@ -216,6 +218,11 @@ def __init__(
         trace_configs: Optional[List[TraceConfig]] = None,
         read_bufsize: int = 2 ** 16,
     ) -> None:
+        if isinstance(base_url, str):
+            self._base_url: Optional[URL] = URL(base_url)
+        else:
+            self._base_url = base_url
```

Comment:
```suggestion
        if base_url is None or isinstance(base_url, URL):
            self._base_url: Optional[URL] = base_url
        else:
            self._base_url = URL(base_url)
```

The reversed check handles more errors in runtime, e.g. passing a number or list as `base_url`

Reply:
Sorry, edited str->URL.
Now the snipped is correct IMHO